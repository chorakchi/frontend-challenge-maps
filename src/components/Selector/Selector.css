.container-selector {
  width: 100%;
  text-align: start;
  line-height: 30px;
  padding: 20px;
  }