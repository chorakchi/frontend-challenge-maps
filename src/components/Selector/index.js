import React from "react";
import PropTypes from "prop-types";
import "./Selector.css";

class Selector extends React.Component {
  state = {
    selected: this.props.defaultValue,
  };

  mapsApiLoaded = null;
  mapInstance = null;

  handlerOnChange = (e) => {
    this.setState({ selected: e.target.value });
    this.props.onChange(e.target.value);
  };

  render() {
    return (
      <div className="container-selector">
        {this.props.label}
        <form>
          {this.props.options.map((option) => (
            <span id={"container-" + option.value} key={option.value}>
              <input
                type="radio"
                id={option.value}
                name="foodOption"
                value={option.value}
                checked={this.state.selected === option.value}
                onChange={this.handlerOnChange}
              />
              <label htmlFor={option.value}>{option.name}</label>
            </span>
          ))}
        </form>
      </div>
    );
  }
}

Selector.propTypes = {
  options: PropTypes.array,
  defaultValue: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func,
};

export default Selector;
