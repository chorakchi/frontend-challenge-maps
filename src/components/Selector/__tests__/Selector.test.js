// @flow

import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Selector from "./../index";

describe("Selector", () => {
  const defaultProps = {
    options: [
      { name: "Pizza", value: "Pizza" },
      { name: "Burger", value: "Burger" },
      { name: "Sushi", value: "Sushi" },
    ],
    defaultValue: "Pizza",
    label: "What would you like to eat?",
    onChange: () => {},
  };

  it("should render the Selector", () => {
    const subject = shallow(<Selector {...defaultProps} />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Selector {...defaultProps} />).toJSON();
    expect(subject).toMatchSnapshot();
  });

  it("rednder correctly the options in selector", () => {
    const mockCallback = jest.fn(() => "i return something!");
    const subject = mount(
      <Selector {...defaultProps} onChange={mockCallback} />
    );
    expect(subject.find("input")).toHaveLength(defaultProps.options.length);
  });
  it("should call onChangeSelector and rednder correctly and have a key on selecting Burger", () => {
    const mockCallback = jest.fn(() => "i return something!");
    const subject = mount(
      <Selector {...defaultProps} onChange={mockCallback} />
    );
    expect(subject.find("#Burger")).toHaveLength(1);
    subject
      .find("#Burger")
      .props()
      .onChange({ target: { value: "Burger" } });
    expect(subject.find("#container-Burger").key()).toEqual("Burger");
    expect(mockCallback).toHaveBeenCalledTimes(1);
    expect(mockCallback).toBeCalledWith("Burger");
    expect(subject.state("selected")).toEqual("Burger");
  });
});
