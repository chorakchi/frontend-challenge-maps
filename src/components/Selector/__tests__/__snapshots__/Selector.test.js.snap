// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector should render correctly and match to the Snapshot 1`] = `
<div
  className="container-selector"
>
  What would you like to eat?
  <form>
    <span
      id="container-Pizza"
    >
      <input
        checked={true}
        id="Pizza"
        name="foodOption"
        onChange={[Function]}
        type="radio"
        value="Pizza"
      />
      <label
        htmlFor="Pizza"
      >
        Pizza
      </label>
    </span>
    <span
      id="container-Burger"
    >
      <input
        checked={false}
        id="Burger"
        name="foodOption"
        onChange={[Function]}
        type="radio"
        value="Burger"
      />
      <label
        htmlFor="Burger"
      >
        Burger
      </label>
    </span>
    <span
      id="container-Sushi"
    >
      <input
        checked={false}
        id="Sushi"
        name="foodOption"
        onChange={[Function]}
        type="radio"
        value="Sushi"
      />
      <label
        htmlFor="Sushi"
      >
        Sushi
      </label>
    </span>
  </form>
</div>
`;
